crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -0,0 +1,964 @@
+    }
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+pub struct TrustpayPaymentsResponse {
+    pub status: Option<i64>,
+    pub description: Option<String>,
+    #[serde(rename = "instanceId")]
+    pub instance_id: Option<String>,
+    #[serde(rename = "paymentStatus")]
+    pub payment_status: Option<String>,
+    #[serde(rename = "paymentDescription")]
+    pub payment_description: Option<String>,
+    #[serde(rename = "redirectUrl")]
+    pub redirect_url: Option<String>,
+    #[serde(rename = "redirectParams")]
+    pub redirect_params: Option<HashMap<String, String>>,
+    #[serde(rename = "PaymentRequestId")]
+    pub payment_request_id: Option<i64>,
+    #[serde(rename = "GatewayUrl")]
+    pub gateway_url: Option<Url>,
+    #[serde(rename = "ResultInfo")]
+    pub payment_result_info: Option<ResultInfo>,
+    #[serde(rename = "PaymentMethod")]
+    pub payment_method_response: Option<TrustpayPaymentMethod>,
+    #[serde(rename = "MerchantIdentification")]
+    pub merchant_identification_response: Option<MerchantIdentification>,
+    #[serde(rename = "PaymentInformation")]
+    pub payment_information_response: Option<BankPaymentInformationResponse>,
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, TrustpayPaymentsResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            TrustpayPaymentsResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let (status, error, payment_response_data) =
+            get_trustpay_response(item.response, item.http_code)?;
+        Ok(Self {
+            status,
+            response: error.map_or_else(|| Ok(payment_response_data), Err),
+            ..item.data
+        })
+    }
+}
+
+pub fn get_trustpay_response(
+    response: TrustpayPaymentsResponse,
+    status_code: u16,
+) -> CustomResult<
+    (
+        enums::AttemptStatus,
+        Option<types::ErrorResponse>,
+        types::PaymentsResponseData,
+    ),
+    errors::ConnectorError,
+> {
+    match (
+        response.instance_id,
+        response.payment_status,
+        response.payment_request_id,
+        response.gateway_url,
+        response.payment_result_info,
+        response
+            .payment_information_response
+            .clone()
+            .unwrap_or_default()
+            .status,
+    ) {
+        (Some(instance_id), Some(payment_status), _, _, _, _) => {
+            let (status, msg) =
+                get_transaction_status(payment_status.as_str(), response.redirect_url.clone())?;
+            let form_fields = response
+                .redirect_params
+                .unwrap_or(std::collections::HashMap::new());
+            let redirection_data = response.redirect_url.map(|url| services::RedirectForm {
```

Comment:
try to use `services::RedirectForm::from(response.redirect_url, services::Method::Post)`, this might solve your use case